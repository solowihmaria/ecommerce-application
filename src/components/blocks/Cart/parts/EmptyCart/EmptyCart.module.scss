@use '../../../../../assets/styles/constants.module' as c;

.emptyCartContent {
    display: flex;
    flex-direction: column;
    gap: 5rem;
    align-items: center;
    justify-content: center;
    height: min-content;
    padding: 3rem 2rem;
    border: 1px solid c.$border;
    border-radius: 1.2rem;
    backdrop-filter: blur(15px);
}

.cartTitle {
    padding: 0 2rem;
    font-size: 2rem;
}

.emptyCartText {
    color: c.$color-text-primary;
}

.button.emptyCartButton {
    width: 100%;
    background-color: c.$color-accent-primary;
}

.emptyCartButtonContainer {
    width: 100%;
    max-width: 32rem;
}
