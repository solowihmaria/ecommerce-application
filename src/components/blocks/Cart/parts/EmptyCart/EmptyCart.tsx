import { Link } from 'react-router-dom';
import { Button } from '../../../../ui/Button';
import styles from './EmptyCart.module.scss';
import { Heading } from '../../../../ui/Heading';
import clsx from 'clsx';

export const EmptyCart = () => {
    return (
        <div className={styles.emptyCartContent}>
            <Heading className={styles.cartTitle}>Shopping Cart</Heading>
            <p className={styles.emptyCartText}>Shopping cart is empty</p>
            <Link to="/catalog" className={styles.emptyCartButtonContainer}>
                <Button className={clsx(styles.emptyCartButton, styles.button)}>
                    Start shopping
                </Button>
            </Link>
        </div>
    );
};
