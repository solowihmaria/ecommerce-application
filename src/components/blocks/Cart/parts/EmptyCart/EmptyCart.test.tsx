import '@testing-library/jest-dom';

import { EmptyCart } from './EmptyCart';
import { MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';

describe('<ErrorCart/> component', () => {
    test('Render ErrorCart', () => {
        render(
            <MemoryRouter>
                <EmptyCart />
            </MemoryRouter>
        );

        const emptyCartContainer = screen.getByTestId('empty-cart-test-id');
        expect(emptyCartContainer).toBeInTheDocument();
        expect(emptyCartContainer).toMatchSnapshot();
    });
});
