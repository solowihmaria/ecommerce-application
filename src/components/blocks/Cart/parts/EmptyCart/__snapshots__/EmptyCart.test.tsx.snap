// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorCart/> component Render ErrorCart 1`] = `
<div
  class="emptyCartContent"
  data-testid="empty-cart-test-id"
>
  <h1
    class="heading heading--h1 cartTitle"
  >
    Shopping Cart
  </h1>
  <p
    class="emptyCartText"
  >
    Shopping cart is empty
  </p>
  <a
    class="emptyCartButtonContainer"
    data-discover="true"
    href="/catalog"
  >
    <button
      class="button primary emptyCartButton button"
    >
      Start shopping
    </button>
  </a>
</div>
`;
